### PR TITLE
fix(Scripts/Spells): Windfury Weapon should not consume Flurry stacks.

### LINKED ISSUE #6234: Windfury Non-Crit Strikes Consume Stacks of Flurry
#### WHAT CLIENT DO YOU PLAY ON?

- [ ] enGB
- [x] enUS
- [ ] other (specify)

##### FACTION

- [x] Alliance
- [x] Horde

##### CONTENT PHASE:

- [x] Generic
- [ ] 1-19
- [ ] 20-29
- [ ] 30-39
- [ ] 40-49
- [ ] 50-59

##### CURRENT BEHAVIOUR:

Original report: https://github.com/chromiecraft/chromiecraft/issues/784

When Windfury Weapon procs while charges of Flurry are present, the stacks of Flurry are consumed.

Tester's notes: It seems a bit more complicated. To summarize, Flurry gives you 3 stacks of a haste buff that is consumed per melee strike. Windfury gives you 2 extra melee strikes when it procs. However, WF procs should not consume Flurry stacks. But WF strikes themselves can proc Flurry if they crit.

What seems to be happening is that WF melee attacks will consume Flurry stacks **if they are non criticals.** WF crits behave correctly and do not consume Flurry stacks.

Here's 6 clips showing this:
Clip 1. We start with 3 x Flurry. The next melee strike procs 2 WF non-crits, making 3 strikes, which instantly consumes the Flurry stack.
Clip 2. We start with 3 x Flurry. The next melee strike procs 1 x WF non-crit and 1 WF crit. 2 Flurry stacks are consumed, leaving 1.
Clip 3. Same as 2.
Clip 4. We start with 2 x Flurry, and 1 x melee and 2 x WF strikes consume the stack.
Clip 5. Same as 2 and 3.
Clip 6. Same as 1, all 3 stacks consumed.

https://user-images.githubusercontent.com/81782124/120884641-63799700-c623-11eb-8537-afd1c2c4294d.mp4

##### EXPECTED BLIZZLIKE BEHAVIOUR:

Windfury Weapon effects should not consume Flurry.

"Windfury procs themselves do not consume Flurry." - https://wowpedia.fandom.com/wiki/Windfury_Weapon?oldid=2348015

##### STEPS TO REPRODUCE THE PROBLEM:

1. Enchant a weapon with Windfury Weapon.
2. Proc Flurry by landing a critical hit.
3. Proc Windfury while stacks of Flurry are present. You need either 3 or 2 stacks of flurry to see anything, because of course if you ony have 1 stack it will be consumed by the next strike anyway.

##### AC HASH/COMMIT:

https://github.com/chromiecraft/azerothcore-wotlk/commit/40cf7b46d221e562a20264316d660318af213690

##### OPERATING SYSTEM:

Ubuntu 20.04

##### MODULES:

- [mod-ah-bot](https://github.com/azerothcore/mod-ah-bot)
- [mod-cfbg](https://github.com/azerothcore/mod-cfbg)
- [mod-desertion-warnings](https://github.com/azerothcore/mod-desertion-warnings)
- [mod-duel-reset](https://github.com/azerothcore/mod-duel-reset)
- [mod-eluna-lua-engine](https://github.com/azerothcore/mod-eluna-lua-engine)
- [mod-multi-client-check](https://github.com/azerothcore/mod-multi-client-check)
- [mod-pvp-titles](https://github.com/azerothcore/mod-pvp-titles)
- [mod-server-auto-shutdown](https://github.com/azerothcore/mod-server-auto-shutdown)
- [lua-CarbonCopy](https://github.com/55Honey/Acore_CarbonCopy)
- [lua-LevelUpReward](https://github.com/55Honey/Acore_LevelUpReward)
- [lua-send-and-bind](https://github.com/55Honey/Acore_SendAndBind)

##### OTHER CUSTOMIZATIONS:

None.

##### SERVER:

ChromieCraft

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/98988592-windfury-non-crit-strikes-consume-stacks-of-flurry?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F40032087&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by UltraNix**
Fixed #6234

## Changes Proposed:
-  Added script to disable consuming Flurry stacks by Windfury Weapon.

## Issues Addressed:

- Closes #6234
- Closes https://github.com/chromiecraft/chromiecraft/issues/784

## Tests Performed:

- Tested ingame.

## How to Test the Changes:

1. Enchant a weapon with Windfury Weapon.
2. Proc Flurry by landing a critical hit.
3. Proc Windfury while stacks of Flurry are present. You need either 3 or 2 stacks of flurry to see anything, because of course if you ony have 1 stack it will be consumed by the next strike anyway.

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

**Comment by pklloveyou** [comment]
It's too complicated. should wait for the author of the error problem to test LOL

File: src/server/scripts/Spells/spell_shaman.cpp [hunk 1]
```diff
@@ -1354,6 +1354,41 @@ class spell_sha_thunderstorm : public SpellScriptLoader
     }
 };
 
+class spell_sha_flurry_proc : public SpellScriptLoader
```
Comment (Kitzunu) [inline src/server/scripts/Spells/spell_shaman.cpp:1357]:
```suggestion
// -16257 - SpellName
class spell_sha_flurry_proc : public SpellScriptLoader
```

**Comment by pklloveyou** [comment]
image
I can't play shaman, black wind doesn't consume the "dance" spell, pass the test?

**Review by Helias — APPROVED** [review]

**Comment by Azcobu** [comment]
> It's too complicated. should wait for the author of the error problem to test LOL

Fair enough. Checked it out and this is working properly now, both WF crits and non-crits are correctly not removing Flurry stacks. So now you'll see a strike proc Flurry and WF at the same time, and the Flurry stack will still have 3 stacks after the WF attacks. So yeah, looks like a good fix, nice job.

WoWScrnShot_073121_212022
WoWScrnShot_073121_212027

**Comment by pklloveyou** [comment]
nice

**Comment by FrancescoBorzi** [comment]
@UltraNix merge conflicts

**Pull request merged**